apps/examples/media_tash/media_tash_main.c

```diff
@@ -0,0 +1,420 @@
+		printf("6) playing volume down\n");
+		printf("7) Forward\n");
+		printf("8) Backward\n");
+		printf("-----------------\n");
+		printf("a) media_record\n");
+		printf("b) media_record_stream\n");
+		printf("c) media_pause_record\n");
+		printf("d) media_resume_record\n");
+		printf("e) media_stop_record\n");
+		printf("f) before record(a and d), play_file_\n");
+		printf("-----------------\n");
+		printf("x) Exit\n");
+		printf(">> ");
+
+		ch = getchar();
+		printf("Instruction: %c\n", ch);
+		switch (ch) {
+		case '0': {
+			ret = network_setup();
+
+			if (ret < 0) {
+				printf("Network Error!!\n");
+				return 0;
+			}
+		}
+		break;
+		case '1': {
+			if (media_is_recording()) {
+				media_stop_record();
+				disconnect_socket(&fd);
+			}
+
+			if (media_play_init() != PLAY_OK) {
+				printf("Error: media_play_init()\n");
+				return 0;
+			}
+			else {
+				printf("MediaPlay Init Okay..\n");
+			}
+
+			printf("Socket connecting...\n");
+			fd = connect_socket("192.168.1.213", 8080);
+			if (fd < 0) {
+				printf("Connection failed...\n");
+				dm_conn_wifi_disconnect();
+				return 0;
+			}
+
+			printf("Socket connected successfully.\n");
+	
+			ret = media_play(fd, MEDIA_FORMAT_WAV);
+			if(ret != MEDIA_OK) {
+				printf("Error: media_play\n");
+				DISCONNECT_WIFI_AND_CLOSE_SOCKET();
+				return 0;
+			}
+		}
+		break;
+		case '2': {
+			ret = media_pause_play();
+			if (ret != MEDIA_OK) {
+				printf("Error: media_pause_play\n");
+			}
+		}
+		break;
+		case '3': {
+			ret = media_resume_play();
+			if (ret != MEDIA_OK) {
+				printf("Error: media_pause_play\n");
+			}
+		}
+		break;
+		case '4': {
+			ret = media_stop_play();
+			if (ret != MEDIA_OK) {
+				printf("Error: media_stop_play\n");
+			}
+		}
+		break;
+		case '5': {
+			if (vol < 255) {
+				vol += 10;
+			}
+
+			ret = media_set_vol(vol);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_vol\n");
+			}
+			else {
+				printf("Current Volume: %d\n", vol);
+			}
+		}
+		break;
+		case '6': {
+			if (vol > 15) {
+				vol -= 10;
+			}
+
+			ret = media_set_vol(vol);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_vol\n");
+			}
+			else {
+				printf("Current Volume: %d\n", vol);
+			}
+		}
+		break;
+		case '7': {
+			pos += 1000;
+			if (pos > media_len - 1) {
+				pos = media_len - 1;
+			}
+
+			ret = media_set_position(pos);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_position\n");
+			}
+		}
+		break;
+		case '8': {
+			pos -= 1000;
+			if (pos < 0) {
+				pos = 0;
+			}
+
+			ret = media_set_position(pos);
+			if (ret != MEDIA_OK) {
+				printf("Error: media_set_position\n");
+			}
+		}
+		break;
+		case 'a': {
+			if (media_is_playing()) {
+				media_stop_play();
+				disconnect_socket(&fd);
+			}
+
+			if (media_record_init() != RECORD_OK) {
+				printf("Error: media_record_init()\n");
+				return 0;
+			}
+			else {
+				printf("MediaRecord Init Okay..\n");
+			}
+ 
+			fd = open("/mnt/record", O_TRUNC | O_CREAT | O_RDWR);
```

Comment:
We should randomize this file name, we do not want to remove a file if it was created by user by this name. 
`
#define FILE_NAME_LENGTH 15
#define RECORD_FILE_NAME "record"
snprintf(filename, FILE_NAME_LENGTH, "%s.%x\0", RECORD_FILE_NAME, (unsigned)(rand() & 0xffff));
`